### PR TITLE
Notification

### PR DESCRIPTION
**Pull request by OneX**

### CODE REVIEW

File: packages/ui/src/01-atoms/to-top/index.tsx [hunk 1]
```diff
@@ -8,25 +8,9 @@ interface Props {
 
 const ToTop: React.FC<Props> = ({ label }) => (
     <Button classes={['a-btn--to-top']} attrs={{ 'data-module': 'to-top' }}>
```
Comment (neno) [inline packages/ui/src/01-atoms/to-top/index.tsx:10]: Nice, you did it!

File: packages/ui/src/02-molecules/tab-list/index.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.m-tablist {
+  @extend .l-divider-medium;
```
Comment (neno) [inline packages/ui/src/02-molecules/tab-list/index.scss:2]: - [ ] don't use @extend

File: packages/ui/src/03-organisms/form/index.scss [hunk 3]
```diff
@@ -12,6 +12,6 @@
   }
 
   &__notification {
-    @extend .l-divider-big;
+    @extend .l-divider-medium;
```
Comment (neno) [inline packages/ui/src/03-organisms/form/index.scss:15]: - [ ] don't use @extend

File: packages/ui/src/04-templates/login-template/index.tsx [hunk 4]
```diff
@@ -10,20 +10,32 @@ import { tabs, loginTemplateData } from './stories/login-template.data';
...
-            </TabPanel>
-            <TabPanel id="pensionierte" active={false}>
-                <LoginForm username={loginTemplateData.username} password={loginTemplateData.password} />
-            </TabPanel>
-        </Tabs>
-
+        <div className="l-divider-medium">
+            <Tabs tabs={tabs} isTabListCentered>
+                <TabPanel id="mitarbeitende" active>
+                    <div className="u-text-center">
```
Comment (neno) [inline packages/ui/src/04-templates/login-template/index.tsx:19]: Why do you need this utility class?

File: packages/ui/src/04-templates/login-template/index.tsx [hunk 5]
```diff
@@ -10,20 +10,32 @@ import { tabs, loginTemplateData } from './stories/login-template.data';
...
+                        <Button classes={['a-btn--primary a-btn--arrows']} arrows url="/">
+                            Login
+                        </Button>
+                    </div>
+                </TabPanel>
+                <TabPanel id="pensionierte" active={false}>
+                    <div className="l-login-panel">
+                        <div className="u-text-center">
```
Comment (neno) [inline packages/ui/src/04-templates/login-template/index.tsx:27]: Same question as above

File: packages/ui/src/js/index.js [hunk 6]
```diff
@@ -11,10 +11,6 @@ const timeout = process.env.STORYBOOK_TIMEOUT || 0;
  */
 
 setTimeout(() => {
-    document.querySelectorAll("[data-module='collapsible']").forEach(el => {
```
Comment (neno) [inline packages/ui/src/js/index.js:14]: don't we need that anymore?

File: packages/ui/src/js/libs/FormValidation/ErrorHandler.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+        message.innerHTML = '';
+        message.style.display = 'none';
+        message.style.visibility = 'hidden';
+    }
+
+    removeInvalidInputAttr(field) {
+        field.classList.remove(this.fieldErrorClass);
+        field.removeAttribute('aria-describedby');
+
```
Comment (neno) [inline packages/ui/src/js/libs/FormValidation/ErrorHandler.js:22]: remove extra line

File: packages/ui/src/js/libs/FormValidation/ErrorHandler.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        }
+    }
+
+    show(field, error) {
+        const id = field.id || field.name;
+        if (!id) return;
+
+        this.createErrorMessage(field, error, id);
+        this.addInvalidInputAttr(field, id);
+
```
Comment (neno) [inline packages/ui/src/js/libs/FormValidation/ErrorHandler.js:66]: remove extra line

File: packages/ui/src/js/libs/FormValidation/ErrorList.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
+export default class ErrorList {
+    constructor(invalidFields, listClass) {
+        this.invalidFields = invalidFields;
+        this.listClass = listClass;
+        this.handleClick = this.handleClick.bind(this);
+
```
Comment (neno) [inline packages/ui/src/js/libs/FormValidation/ErrorList.js:6]: remove extra line

File: packages/ui/src/js/libs/FormValidation/index.js [hunk 10]
```diff
@@ -106,19 +108,40 @@ const Tabs = createModule({
...
+
         // Public Methods
         state.init = () => {
             disableNativeValidation();
             bindEvents();
         };
 
         state.destroy = () => {
-            unbindEvents();
+            destroy();
```
Comment (neno) [inline packages/ui/src/js/libs/FormValidation/index.js:139]: use methods directly:

```
unbindEvents();
destroyNotification();
```

File: packages/ui/src/js/libs/Notification.js [hunk 11]
```diff
@@ -1,90 +1,87 @@
...
+        state.append = node => {
+            append(node);
+        };
+
+        state.init = () => {
+            init();
+        };
+
+        state.destroy = () => {
+            destoryNotification();
```
Comment (neno) [inline packages/ui/src/js/libs/Notification.js:79]: typo "destory"

File: packages/ui/src/js/libs/Notification.js [hunk 12]
```diff
@@ -1,90 +1,87 @@
...
+            addTextContainer();
+
+            el.insertBefore(notification, el.firstChild);
+        };
+
+        const init = () => {
+            createNotification();
+        };
+
+        const destoryNotification = () => {
```
Comment (neno) [inline packages/ui/src/js/libs/Notification.js:64]: typo "destory"

File: packages/ui/src/js/libs/Tabs.js [hunk 13]
```diff
@@ -56,32 +59,35 @@ const Tabs = createModule({
...
+                    } else if (key === 'ArrowRight' || key === 'Right') {
                         focusFirstTab();
                     }
                 }
             }
         };
 
         const handleKeyup = event => {
-            const key = event.which || event.keyCode;
+            const key = event.key || event.keyCode;
```
Comment (neno) [inline packages/ui/src/js/libs/Tabs.js:80]: why do you need keyCode?

File: packages/ui/src/styles/base/_typo.scss [hunk 14]
```diff
@@ -1,27 +1,55 @@
+@mixin block-margin {
+  margin: 0 0 rem(20) 0;
```
Comment (neno) [inline packages/ui/src/styles/base/_typo.scss:2]: Shouldn't that also be em()?

File: packages/ui/src/styles/base/_typo.scss [hunk 15]
```diff
@@ -1,27 +1,55 @@
...
+  @include media-breakpoint-up(tablet) {
+    font-size: $font-size-h2-tablet-up;
+  }
 }
 
 h3,
 .h3 {
-  @extend %h-3;
+  font-size: $font-size-h3;
+  font-weight: 300;
```
Comment (neno) [inline packages/ui/src/styles/base/_typo.scss:34]: use variable?

File: packages/ui/src/styles/base/_typo.scss [hunk 16]
```diff
@@ -1,27 +1,55 @@
...
+  @include media-breakpoint-up(mobile) {
+    font-size: $font-size-h3-mobile-up;
+  }
 }
 
 h4,
 .h4 {
-  @extend %h-4;
+  font-size: 1em;
+  font-weight: 700;
```
Comment (neno) [inline packages/ui/src/styles/base/_typo.scss:45]: use variable?

File: packages/ui/src/styles/tools/_helpers.scss [hunk 17]
```diff
@@ -1,7 +1,3 @@
-.sr-only {
-  @include hide-visually;
-}
-
 .u {
   &-text-center {
```
Comment (neno) [inline packages/ui/src/styles/tools/_helpers.scss:2]: why do we need this?

File: packages/ui/src/styles/tools/_helpers.scss [hunk 18]
```diff
@@ -14,4 +10,8 @@
   &-text-right {
```
Comment (neno) [inline packages/ui/src/styles/tools/_helpers.scss:10]: why do we need this?

**Review by neno — CHANGES_REQUESTED** [review]

File: packages/ui/src/styles/base/_typo.scss (hunk 14 above)
Comment (OneX) [reply packages/ui/src/styles/base/_typo.scss:2]: I am not sure for this one right now. I did not check how they separate paragraph.

File: packages/ui/src/js/libs/FormValidation/index.js (hunk 10 above)
Comment (OneX) [reply packages/ui/src/js/libs/FormValidation/index.js:139]: I am using destroy function within form validation itself, on line 81.

**Pull request merged**